src/dialog/provider.js

```diff
@@ -18,107 +18,72 @@ export default class FlowProvider extends EventEmitter2 {
     this.logger = logger
     this.botfile = botfile
     this.projectLocation = projectLocation
+    this.ghostManager = ghostManager
+    this.flowsDir = this.botfile.flowsDir || './flows'
+
+    mkdirp.sync(path.dirname(this.flowsDir))
+    this.ghostManager.addRootFolder(this.flowsDir, '**/*.json')
   }
 
   async loadAll() {
-    const relDir = this.botfile.flowsDir || './flows'
-    const flowsDir = path.resolve(this.projectLocation, relDir)
-
-    if (!fs.existsSync(flowsDir)) {
-      return []
-    }
-
-    const searchOptions = { cwd: flowsDir }
-
-    const flowFiles = await Promise.fromCallback(callback => glob('**/*.flow.json', searchOptions, callback))
-
-    const flows = []
+    const flowFiles = await this.ghostManager.directoryListing(this.flowsDir, '.flow.json')
 
-    flowFiles.forEach(file => {
-      const filePath = path.resolve(flowsDir, './' + file)
-      const flow = JSON.parse(fs.readFileSync(filePath))
+    const flows = await Promise.all(
+      flowFiles.map(async name => {
+        const uiFileName = name.replace(/\.flow/g, '.ui')
```

Comment:
better (safer) do it explicitly: `name.replace(/\.flow\.json$/, '.ui.json')`

Reply:
Refactored that.